src/overlays/actors/ovl_Obj_Etcetera/z_obj_etcetera.h

```diff
@@ -3,13 +3,32 @@
 
 #include "global.h"
 
+#define OBJETCETERA_TYPE(thisx) (((thisx)->params & 0xFF80) >> 7)
+
+typedef enum {
+    /*  0   */ TYPE_PINK_FLOWER,
+    /*  1   */ TYPE_PINK_FLOWER_SPAWNED_FROM_MAD_SCRUB,
+    /*  2   */ TYPE_GOLD_FLOWER,
+    /*  3   */ TYPE_GOLD_FLOWER_SPAWNED_FROM_MAD_SCRUB,
+} ObjEtceteraType;
+
+#define NUMBER_OF_FLOWER_TYPES 4
+
```

Comment:
```suggestion
typedef enum {
    /* 0 */ DEKU_FLOWER_TYPE_PINK,
    /* 1 */ DEKU_FLOWER_TYPE_PINK_SPAWNED_FROM_MAD_SCRUB,
    /* 2 */ DEKU_FLOWER_TYPE_GOLD,
    /* 3 */ DEKU_FLOWER_TYPE_GOLD_SPAWNED_FROM_MAD_SCRUB,
    /* 4 */ DEKU_FLOWER_TYPE_MAX,
} ObjEtceteraType;
```
This is what we usually do is such circumstances: add a meximum enum to compare with (and use it in the code instead of the define). These are in the header so globally exposed, so they need a unique name. And there's no need for so much spacing around the numbers when they're only single digits.